pkg/capture/flow.go

```diff
@@ -413,55 +394,42 @@ func (fs FlowInfos) TablePrint(w io.Writer) error {
 	return tw.Flush()
 }
 
-func (f *Flow) updateDirection(epHash capturetypes.EPHash, auxInfo byte) {
-	if direction := capturetypes.ClassifyPacketDirection(epHash, f.isIPv4, auxInfo); direction != capturetypes.DirectionUnknown {
-		f.directionConfidenceHigh = direction.IsConfidenceHigh()
+// Byte-level lookup table for common ports (allows for constant-time lookup that is almost as
+// fast as a best case conditional logic)
+var commonPorts = [18][32][256]bool{
```

Comment:
It's a rather `C`ish thing to do, but I picked that up when looking for optimized data structures / flows in some blog article and even with sparse data like here you can very easily sacrifice a couple of kiB of your heap to achieve constant-time lookup behavior for whatever you need / like. IMHO it can also produce much more readable code than a nested `if/then/else` BS (especially if _that_ has to be optimized)...